miden-lib/src/transaction/memory.rs

```diff
@@ -10,19 +10,34 @@ pub type StorageSlot = u8;
 // PUBLIC CONSTANTS
 // ================================================================================================
 
+// General layout
+//
 // | Section           | Start address |  End address |
 // | -------------     | :------------:| :-----------:|
-// | Bookkeeping       | 0             | 4            |
+// | Bookkeeping       | 0             | 6            |
 // | Global inputs     | 100           | 105          |
 // | Block header      | 200           | 208          |
 // | Chain MMR         | 300           | 332?         |
-// | Account data      | 400           | 651?         |
-// | Account storage   | 499           | ?            |
-// | Account procedures| 1499          | ?            |
-// | Kernel data       | 9_999         | 10_028?      |
+// | Kernel data       | 400           | 429          |
+// | Accounts data     | 2048          | 133_119      | 64 foreign accounts max
 // | Input notes       | 1_048_576     | ?            |
 // | Output notes      | 4_194_304     | ?            |
 
+// Relative layout of one account
+//
+// | Section           | Start pointer |  End pointer |
+// | -------------     | :------------:| :-----------:|
+// | Id and nonce      | 0             | 0            |
+// | Vault root        | 1             | 1            |
+// | Storage root      | 2             | 2            |
+// | Code root         | 3             | 3            |
+// | Padding           | 4             | 6            |
+// | Num procedures    | 7             | 7            |
+// | Procedures info   | 8             | 519          |
+// | Num storage slots | 520           | 520          |
+// | Storage slot info | 521           | 1030         |
```

Comment:
Let's start storage slot info on an even address. Maybe we move "num storage slots" to 521 and "storage slot info" to 522?